syntax = "proto3";

option go_package = "./proto";

service Review {
    rpc ProcessReview (MyReview) returns (MyResponse) {};
}

message MyResponse {
    bool saved = 1;
}

enum Rating {
    EXCELLENT = 0;
    GOOD = 1;
    OK = 2;
    MEH = 3;
    BOGUS = 4;
}

message MyReview {
    uint64 userid = 1;
    uint64 timestamp = 2;
    optional string message = 3;
    optional Rating rating = 4;
}

